internal/tiltfile/live_update.go

```diff
@@ -181,66 +221,51 @@ func (s *tiltfileState) liveUpdateStepToModel(l liveUpdateStep) (model.LiveUpdat
 	}
 }
 
-func (s *tiltfileState) liveUpdateToModel(l liveUpdateDef) (model.LiveUpdate, error) {
-	return model.NewLiveUpdate(l.steps, model.NewPathSet(l.fullRebuildTriggers, s.absWorkingDir()))
-}
-
-func (s *tiltfileState) consumeLiveUpdateStep(stepToConsume liveUpdateStep) {
-	for i, step := range s.unconsumedLiveUpdateSteps {
-		if step.declarationPosition() == stepToConsume.declarationPosition() {
-			copy(s.unconsumedLiveUpdateSteps[i:], s.unconsumedLiveUpdateSteps[i+1:])
-			s.unconsumedLiveUpdateSteps = s.unconsumedLiveUpdateSteps[:len(s.unconsumedLiveUpdateSteps)-1]
-			break
-		}
-	}
-}
-
-func (s *tiltfileState) liveUpdate(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
-	var dockerRef string
-	var steps, fullRebuildTriggers starlark.Value
-	if err := starlark.UnpackArgs(fn.Name(), args, kwargs,
-		"image", &dockerRef,
-		"steps", &steps,
-		"full_rebuild_triggers?", &fullRebuildTriggers,
-	); err != nil {
-		return nil, err
-	}
-
+func (s *tiltfileState) liveUpdateFromSteps(maybeSteps starlark.Value) (model.LiveUpdate, error) {
 	var modelSteps []model.LiveUpdateStep
-	stepSlice := starlarkValueOrSequenceToSlice(steps)
+	stepSlice := starlarkValueOrSequenceToSlice(maybeSteps)
+	var fallBackOn []string
+
 	for _, v := range stepSlice {
 		step, ok := v.(liveUpdateStep)
 		if !ok {
-			return starlark.None, fmt.Errorf("'steps' must be a list of live update steps - got value '%v' of type '%s'", v.String(), v.Type())
+			return model.LiveUpdate{}, fmt.Errorf("'steps' must be a list of live update steps - got value '%v' of type '%s'", v.String(), v.Type())
 		}
 
 		ms, err := s.liveUpdateStepToModel(step)
 		if err != nil {
-			return starlark.None, err
+			return model.LiveUpdate{}, err
 		}
 		s.consumeLiveUpdateStep(step)
-		modelSteps = append(modelSteps, ms)
-	}
 
-	frtSlice := starlarkValueOrSequenceToSlice(fullRebuildTriggers)
-	var frtStrings []string
-	for _, v := range frtSlice {
-		str, ok := v.(starlark.String)
-		if !ok {
-			return starlark.None, fmt.Errorf("'full_rebuild_triggers' must only contain strings - got value '%v' of type '%s'", v.String(), v.Type())
+		// HACK(maia): temporary hack for backwards compatibility/making this a smaller PR--
+		// use the existing LiveUpdate constructor. (Soon, we'll treat fall_back_on as
+		// just another step INTERNALLY too (instead of as a non-step property of a LiveUpdate).
+		if fallBackStep, ok := ms.(model.LiveUpdateFallBackOnStep); ok {
+			if len(fallBackOn) != 0 {
+				return model.LiveUpdate{}, fmt.Errorf("live_update: cannot specify more than one "+
```

Comment:
I was surprised by this and wondered why we only support specifying one path to fall back on!

After a few seconds' thought, I realized `fall_back_on` probably takes a union of str or list[str]. Maybe worth some text here like "just pass all paths to a single step!"?

(or maybe we should support having multiple? that might look cleaner than a long list of strings in a single step?)

Reply:
ehh, supporting multiple is gonna get weird when actually store them as steps on the `model.LiveUpdate`. I'll fix the error message.

Reply:
actually we do the same thing for adds vs runs don't we. kk